src/main/scala/tech/allegro/marathon/plugin/auth/BasicAuthCredentialsExtractor.scala

```diff
@@ -0,0 +1,17 @@
+package tech.allegro.marathon.plugin.auth
+
+import java.nio.charset.Charset
+
+import org.apache.shiro.codec.Base64
+
+class BasicAuthCredentialsExtractor {
+
+  private val BasicLength = "Basic".length()
+
+  def extractCredentials(authCredentials: Seq[String]) = {
+    authCredentials.headOption.map(_.substring(BasicLength).trim())
+      .map(Base64.decode)
```

Comment:
it would be more efficient (and likely more readable) to merge these 3 maps into one